docs/vgclog_examples.md

```diff
@@ -0,0 +1,594 @@
+|concurrent mark      |intermediate |concurrent                     | `<gc-start>`, `<gc-end>`     |`<concurrent-trace-info>` records the progress of the concurrent mark increment|
+|final collection     |final        | STW              | `<concurrent-global-final>`  |Triggered when the card-cleaning threshold is reached. Child element is `<concurrent-trace-info reason=””>`. Operations and sub-operations include a final concurrent mark, a sweep, and an optional class unload and compact.|
+
+The global GC cycle follows a general structure in the verbose GC log as shown. The lines are indented to help illustrate the flow and attributes and some child elements are omitted for clarity. The structure includes a partial GC cycle that starts and finishes within the global cycle:
+
+```xml
+
+<concurrent-kickoff/>       (global cycle 1st increment recorded)
+
+<exclusive-start/>          (STW pause starts)
+
+<cycle-start/>              (global cycle starts)
+
+<exclusive-end/>            (STW pause ends)
+
+
+(global cycle 2nd increment running concurrently)
+
+
+<exclusive-start/>          (STW for partial GC cycle starts)
+
+
+...                         (partial GC cycle starts and completes)
+
+
+<exclusive-end/>            (STW for partial GC cycle ends)
+
+<exclusive-start/>          (STW pause starts)
+
+<concurrent-global-final/>  (global cycle final increment recorded)
+
+<gc-start/>                 (global cycle final increment starts)
+
+<allocation-stats/>
+
+<mem-info>                  (memory status before operations)
+
+<mem></mem>                 (status of different types of memory)
+
+</mem-info>
+
+</gc-start>
+
+<gc-op> “type=rs-scan"</gc-op>        (roots scanning completed)
+
+<gc-op>”type=card-cleaning" </gc-op>  (card cleaning completed)
+
+<gc-op> “type=mark”</gc-op>           (mark operation completed)
+
+<gc-op> “type=classunload”</gc-op>    (class unload operation completed)
+
+<gc-op ”type=sweep” />                (sweep operation completed)
+
+<gc-end>                     (global cycle final increment ends)
+
+<mem-info>                   (memory status after operations)
+
+<mem></mem>                  (status of different types of memory)
+
+</mem-info>
+
+</gc-end> 
+
+</cycle-end>                 (global cycle ends)
+
+<exclusive-end>              (STW pause ends)
+
+<exclusive-start>            (STW pause starts)
+...
+
+```
+
+The first activity in the cycle is recorded by a `<concurrent-kickoff>` element, which records the start of the first of three increments that make up a `gencon` Global GC cycle. The `<concurrent-kickoff>` element records:
+
+- the reason why the GC cycle was triggered. For a `gencon` global cycle, the cycle is triggered when the amount of available memory reaches a particular threshold value.
+- The target number of bytes the cycle aims to reclaim in the heap.
```

Comment:
As mentioned earlier targetBytes is what what we aim to reclaim, but what we aim to trace concurrently. Typically we would be able to do it (and will stop concurrent increment due to card cleaning threashold). If so, the rest of tracing/marking will be done in later in final STW increment.

Reply:
**Is this correct?**

The first activity in the cycle is recorded by a `<concurrent-kickoff>` element, which records the start of the first of three increments that make up a `gencon` Global GC cycle. The `<concurrent-kickoff>` element records:

- the reason why the GC cycle was triggered. For a `gencon` global cycle, the cycle is triggered when the amount of available memory reaches a particular threshold value, the `thresholdFreeBytes`. The `remainingFreeMemory` value is 329, which is slightly below the `thresholdFreeBytes`. The most recent memory allocation request has triggered this GC cycle.  
- The target number of bytes, `targetBytes`, that the cycle aims to mark.
- The current available memory in the different parts of the heap.

```xml
<concurrent-kickoff id="12362" timestamp="2020-10-18T13:35:44.341">

<kickoff reason="threshold reached" targetBytes="239014924" thresholdFreeBytes="33024922" remainingFree="32933776" tenureFreeBytes="42439200" nurseryFreeBytes="32933776" />

</concurrent-kickoff>
```
The `targetBytes` attribute shows that the cycle aims to trace 239MB (239014924). If the cycle is interrupted by another card cleaning threshold value during the intermediate concurrent increment, the final increment will terminate before the tracing of all 239MB is complete.

**AND LATER ON:**

``xml
<exclusive-start id="12378" timestamp="2020-10-18T13:35:44.594" intervalms="12.075">

<response-info timems="0.108" idlems="0.040" threads="3" lastid="00000000018D3800" lastname="LargeThreadPool-thread-33" />

</exclusive-start>

<concurrent-global-final id="12379" timestamp="2020-10-18T13:35:44.594" intervalms="516905.029" >

<concurrent-trace-info reason="card cleaning threshold reached" tracedByMutators="200087048" tracedByHelpers="12164180" cardsCleaned="4966" workStackOverflowCount="0" />

</concurrent-global-final>
```

The `reason` attribute of the `<concurrent-trace-info>` child element indicates that this final STW increment of this global cycle has been triggered because a card-cleaning threshold has been reached. The concurrent tracing has been terminated prematurely and the `targetBytes` tracing target, recorded at the cycle start by `<concurrent-kickoff>`,  has not been achieved. If the concurrent tracing completes without interruption, the `<concurrent-trace-info` element logs `reason=tracing completed`.